packages/web3/src/index.ts

```diff
@@ -70,6 +73,29 @@ export class Web3 extends Web3Context<EthExecutionAPI> {
 
 		const eth = self.use(Eth);
 
+		function ContractBuilder<Abi extends ContractAbi>(
+			this: typeof ContractBuilder,
+			jsonInterface: Abi,
+			address?: Address,
+			options?: ContractInitOptions,
+		) {
+			if (!(this instanceof ContractBuilder)) {
+				throw new ContractError('Please use the "new" keyword to instantiate a contract.');
+			}
+
+			const contract = self.use(Contract, jsonInterface, address, options);
+
+			self._contracts.push(contract);
+
+			return contract;
+		}
+
+		ContractBuilder.setProvider = (_provider: SupportedProviders<EthExecutionAPI>) => {
```

Comment:
currently contract is using universal provider coming via `Web3Context` > `Web3RequestManager` why do we need to explicitly cascade change after once it set at start.

Reply:
To preserve the behaviour of 1.x that once the provider is changed on `Contract` static level, provider must be changed on all instances of any contract.